src/app/code/community/Hackathon/PromoCodeMessages/Model/Validator.php

```diff
@@ -170,6 +170,15 @@ protected function _validateGeneral($rule, $coupon)
             }
         }
 
+        // magemail coupon-level auto-expiration date
+        if ($coupon->getdata('magemail_expired_at') && $coupon->getUsageLimit() && $coupon->getTimesUsed() >= $coupon->getUsageLimit()) {
+            $expirationDate = new Zend_Date($coupon->getdata('magemail_expired_at'), Varien_Date::DATE_INTERNAL_FORMAT);
```

Comment:
`Zend_Date` doesn't take time zone offset for stores into account; only the PHP server timezone. Please use the Mage core date plugin.